It is not up for upstream to decide the valid build type.

diff --git a/ThirdParty/eigen/vtkeigen/CMakeLists.txt b/ThirdParty/eigen/vtkeigen/CMakeLists.txt
index fc5ca90..c9e6a35 100644
--- a/ThirdParty/eigen/vtkeigen/CMakeLists.txt
+++ b/ThirdParty/eigen/vtkeigen/CMakeLists.txt
@@ -21,13 +21,6 @@ if (NOT CMAKE_BUILD_TYPE)
   set(CMAKE_BUILD_TYPE "Release")
 endif()
 
-string(TOLOWER "${CMAKE_BUILD_TYPE}" cmake_build_type_tolower)
-if(    NOT cmake_build_type_tolower STREQUAL "debug"
-   AND NOT cmake_build_type_tolower STREQUAL "release"
-   AND NOT cmake_build_type_tolower STREQUAL "relwithdebinfo")
-  message(FATAL_ERROR "Unknown build type \"${CMAKE_BUILD_TYPE}\". Allowed values are Debug, Release, RelWithDebInfo (case-insensitive).")
-endif()
-
 
 #############################################################################
 # retrieve version infomation                                               #
